openwisp_controller/config/base/config.py

```diff
@@ -459,6 +459,13 @@ def _check_changes(self):
             'backend', 'config', 'context', 'status'
         ).get(pk=self.pk)
         for attr in ['backend', 'config', 'context']:
+            if attr == 'backend' and getattr(self, attr) != getattr(current, attr):
+                config_backend_changed.send(
+                    sender=self.__class__,
+                    instance=self,
+                    old_backend=current.backend,
+                    backend=self.backend,
+                )
```

Comment:
Known bug: signal call is successfull but the result is not coming as expected due to post_save_clear. Working on fixing this bug.

Reply:
it will be better apply templates directly from the model class for a device in case of backend change because m2m_change is also calling after adding new templates. So we can remove this signal and apply devicegroup templates directly to the device on config backend change.